include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -237,34 +225,36 @@ namespace internal
         (Eval::n_rows_of_product > Eval::n_columns_of_product ?
            Eval::n_rows_of_product :
            Eval::n_columns_of_product);
-    Number *temp1 = scratch_data;
+    Number *temp1 = eval.get_scratch_data().begin();
     Number *temp2;
     if (temp_size == 0)
       {
-        temp2 = temp1 + std::max(Utilities::fixed_power<dim>(
-                                   shape_info.data.front().fe_degree + 1),
-                                 Utilities::fixed_power<dim>(
-                                   shape_info.data.front().n_q_points_1d));
+        temp2 =
+          temp1 +
+          std::max(Utilities::fixed_power<dim>(shape_data[0].fe_degree + 1),
+                   Utilities::fixed_power<dim>(shape_data[0].n_q_points_1d));
```

Comment:
what is the reason for replacing `front()` by `[0]`. Personally, I like `front()` because it quickly gives me the hint that code is accessing `ShapeInfo::data` and this is a code - still - not working for anisotropic elements^^